File: python/strings.py
#
# ! Strings are not mutable
sentence = "I love pizza"
# sentence[0] = "i"

# * "len"
print("len(...):", len(sentence))

# * f-strings (solve the problem of concatenation)
print(f'\nf"...{{x}}...": {sentence}')

# * "lower", "upper" (converts the string to lowercase/uppercase)
print("\n.lower(): " + sentence.lower(), "\t\t.upper(): " + sentence.upper())

# * "replace" (replaces a substring with another)
# ! It replaces all occurrences unlike other languages
print('\n.replace("z", "s"): ' + sentence.replace("z", "s"))

# * "strip" (removes whitespace from both ends by default) / "lstrip", "rstrip"
# ^ It can also take an argument to remove a specific character
print('\n.strip(): "  wow  " --> ' + "  wow  ".strip())
# ! "removeprefix", "removesuffix" (they only remove once)

# * "capitalize", "title", "swapcase"
print(
    "\n.capitalize(): " + sentence.capitalize(),
    "\t.title(): " + sentence.title(),
    "\t\t.swapcase(): " + sentence.swapcase(),
)

# * "split" (divide a string into a list of substrings using spaces by default)
# ^ It can also take a character to split the string
print("\n.split():", str(sentence.split()))
# ! However, you can't pass an empty string, use "list" instead

File: javascript/strings.js
let sentence = " I love italian food";
console.log("length:", sentence.length);

//* "indexOf" (the first occurrence of a substring)
//~ "lastIndexOf"
console.log(sentence.indexOf("love"));

//* "slice" (returns a part of a string)
//! It can take negative values as well, solving the problem of "substring"
console.log(sentence.slice(0, 10));

//* "toUpperCase" / "toLowerCase" (converts the string to uppercase / lowercase)
console.log(sentence.toUpperCase(), sentence.toLowerCase());

//* "replace" (replaces the first occurrence of a substring)
console.log(sentence.replace("love", "hate"));

//* "split" (divides a string using a separator)
console.log(sentence.split(" "));

//* "trim" (removes whitespace from both ends)
console.log(sentence.trim());

//* "includes"
console.log(sentence.includes("love"));

//* "repeat"
console.log(sentence.repeat(4));
